services/galaxytoolshed/galaxy-toolshed-base.js

```diff
@@ -0,0 +1,56 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const orderedInstallableRevisionsSchema = Joi.array()
+  .items(Joi.string())
+  .required()
+
+export default class BaseGalaxyToolshedService extends BaseJsonService {
+  static defaultBadgeData = { label: 'galaxy-toolshed' }
+  static baseUrl = 'https://toolshed.g2.bx.psu.edu'
+
+  async fetchOrderedInstallableRevisionsSchema({ repository, owner }) {
+    return this._requestJson({
+      schema: orderedInstallableRevisionsSchema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_ordered_installable_revisions?name=${repository}&owner=${owner}`,
+    })
+  }
+
+  async fetchRepositoryRevisionInstallInfoSchema({
+    repository,
+    owner,
+    schema,
+    changesetRevision,
+  }) {
+    return this._requestJson({
+      schema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_repository_revision_install_info?name=${repository}&owner=${owner}&changeset_revision=${changesetRevision}`,
+    })
+  }
+
+  async fetchLastOrderedInstallableRevisionsSchema({
+    repository,
+    owner,
+    schema,
+  }) {
+    const changesetRevisions =
+      await this.fetchOrderedInstallableRevisionsSchema({
+        repository,
+        owner,
+      })
+    return this.fetchRepositoryRevisionInstallInfoSchema({
+      repository,
+      owner,
+      schema,
+      changesetRevision: changesetRevisions.shift(),
+    })
+  }
+
+  static filterRepositoryRevisionInstallInfo({ response }) {
+    return response
+      .filter(function (x) {
+        return Object.keys(x).length > 0
+      })
+      .shift()
+  }
```

Comment:
I think we can likely get rid of this function entirely, and I'll explain part of why that's the case here as well as on the schema that defines this object which is (currently, but probably shouldn't be) defined in one of the child classes.

Fundamentally I feel like you had this before to aid in some processing which was required based on how you were defining/utilizing the schemas. However, as alluded to in the above comment on the `fetchRepositoryRevisionInstallInfoSchema` definition, we most likely need to change that anyway which should obviate the need for this function.

Reply:
Yes, I forgot this stuff. This is managed by the schema definition. 7a37735b6